clippy_utils/src/macros.rs

```diff
@@ -425,6 +425,33 @@ pub fn find_format_arg_expr<'hir, 'ast>(
     .ok_or(&target.expr)
 }
 
+/// Finds and extracts all format arguments from a format-like macro call.
+///
+/// ```ignore
+/// // vvvvv any format-like macro
+/// println!("Hello, {}!", "ferris")
+/// //                     ^^^^^^^^ returns these expressions
+/// ```
+pub fn collect_format_args<'hir>(
+    cx: &LateContext<'_>,
+    start: &'hir Expr<'hir>,
+    expn_id: ExpnId,
+) -> Vec<&'hir Expr<'hir>> {
+    let mut args = vec![];
+    find_format_args(cx, start, expn_id, |format_args| {
```

Comment:
Since lints would likely want to access the `format_args` part too it would be good to have this API take a callback that has the found args as the second argument. That way we don't have to `find_format_args` twice

Reply:
Done